plugin.php

```diff
@@ -460,6 +460,41 @@ function rest_validate_request_arg( $value, $request, $param ) {
 			}
 		}
 
+		if ( 'array' === $args['type'] ) {
+			// A comma-separated array of IDs could be acceptable instead of an array.
+			if ( is_string( $value ) && isset( $args['allow_csv'] ) && $args['allow_csv'] ) {
+				$value = wp_parse_id_list( $value );
+				if ( empty( $value ) ) {
+					return new WP_Error( 'rest_invalid_param', sprintf( __( '%s requires type %s, or comma-separated list of IDs.', $param, 'array' ) ) );
+				}
+			} elseif ( ! is_array( $value ) ) {
+				return new WP_Error( 'rest_invalid_param', sprintf( __( '%s is not of type %s' ), $param, 'array' ) );
+			}
+
+			// Validate that the items in the array are of the correct type, defaulting to string as the type.
+			$type = isset( $args['items']['type'] ) ? $args['items']['type'] : 'string';
+			$function = '__return_false';
+			switch ( $type ) {
+				case 'integer':
+					$function = 'is_numeric';
+					break;
+
+				case 'boolean':
+					$function = 'rest_is_boolean';
+					break;
+
```

Comment:
I wouldn't worry about supporting boolean because there are no endpoint parameters that use an array of booleans, this PR won't handle 100% properly so let's just not worry about booleans.